StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/STPElementsSessionTest.swift

```diff
@@ -113,6 +113,77 @@ class STPElementsSessionTest: XCTestCase {
         }))
     }
 
+    func testFailedCPMParsing() {
+        // If STPElementsSession decodes a dict...
+        var elementsSessionJson = STPTestUtils.jsonNamed("ElementsSession")!
+        // ...that contains unparseable custom_payment_method_data
+        elementsSessionJson["custom_payment_method_data"] = [
+            "this dict doesn't match the expected shape": "and will fail to parse",
+        ]
+        let elementsSession = STPElementsSession.decodedObject(fromAPIResponse: elementsSessionJson)!
+        // ...it should successfully decode...
+        XCTAssertNotNil(elementsSession)
+        // ...with an empty `customPaymentMethods` property
+        XCTAssertTrue(elementsSession.customPaymentMethods.isEmpty)
+        // ...and send a failure analytic
+        let analyticEvents = STPAnalyticsClient.sharedClient._testLogHistory
+        XCTAssertTrue(analyticEvents.contains(where: { dict in
+            (dict["event"] as? String) == STPAnalyticEvent.paymentSheetElementsSessionCPMLoadFailed.rawValue
+        }))
+    }
+
+    func testSuccessCPMParsing() {
+        // If STPElementsSession decodes a dict...
+        var elementsSessionJson = STPTestUtils.jsonNamed("ElementsSession")!
+
+        let customPaymentMethodsArray: [[String: Any]] = [
+            [
+                "logo_url": "stripe.com",
+                "display_name": "BufoPay (test)",
+                "type": "cpmt_123",
+                "error": NSNull(),
+                "is_preset": false,
+            ],
+            [
+                "logo_url": NSNull(),
+                "display_name": NSNull(),
+                "type": "cpmt_invalid",
```

Comment:
Not your fault, but i think this object was modeled awkwardly.

Reply:
Yeah I agree! This object will be "translated" into something that is displayable in MPE, so we'll have a better model of this in a future PR, but for this type I want it to match the backend behavior.

Totally agree the model of the object in the error state is awkward.